src/components/TransactionForms/CustomForms/Deposit.tsx

```diff
@@ -19,12 +26,75 @@ const Deposit: React.FC<PropsWithChildren<IContractProps>> = ({
   handleFormSubmit,
 }) => {
   const { handleSubmit, watch } = useFormContext<FormData>();
+  const [currencyValue, setCurrencyValue] = useState<string>('');
+  const [currencyIsValid, setCurrencyIsValid] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+
   const depositType: number = watch('depositType');
 
   const onSubmit = async (data: FormData) => {
     await handleFormSubmit(data);
   };
 
+  useEffect(() => {
+    const validateCurrency = async (): Promise<void> => {
+      if (currencyValue === '') {
+        setCurrencyIsValid(false);
+        setError('Invalid Currency');
+        return;
+      }
+
+      setIsLoading(true);
+      setError('');
+
+      try {
+        const fetchResponse = fetch(
+          `${
+            process.env.DEFAULT_API_HOST || 'https://api.testnet.klever.finance'
+          }/assets/${currencyValue}`,
+          {
+            method: 'GET',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+          },
+        );
+
+        const res = await fetchResponse;
+        const data = await res.json();
+
+        if (data?.data?.asset?.assetType === 'Fungible') {
+          setCurrencyIsValid(true);
+          setError('');
+        } else {
+          setCurrencyIsValid(false);
+          setError('Invalid Currency');
+        }
+      } catch (err) {
+        setError('Error validating currency');
+        setCurrencyIsValid(false);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    const timeoutId = setTimeout(validateCurrency, 500);
```

Comment:
increase debounce value to at least 1 sec

Reply:
Done!